src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,608 @@
+            }
+        }
+
+        /// <summary>
+        /// Helper method to validate input data for the relationship fields present in the input for the current entity.
+        /// </summary>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="entityName">Current entity's name.</param>
+        /// <param name="context">Middleware context.</param>
+        /// <param name="objectFieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="fieldsToSupplyToReferencingEntities">Dictionary storing the mapping from relationship name to the set of
+        /// referencing fields for all the relationships for which the current entity is the referenced entity.</param>
+        /// <param name="fieldsToDeriveFromReferencedEntities">Dictionary storing the mapping from relationship name
+        /// to the set of referenced fields for all the relationships for which the current entity is the referencing entity.</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request. As we go down in the multiple mutation from the source
+        /// entity input to target entity input, the nesting level increases by 1. This helps us to throw meaningful exception messages to the user
+        /// as to at what level in the multiple mutation the exception occurred.</param>
+        /// <param name="sqlMetadataProviderFactory">Metadata provider factory.</param>
+        private static void ValidateRelationshipFields(
+            InputObjectType schemaObject,
+            string entityName,
+            IMiddlewareContext context,
+            IReadOnlyList<ObjectFieldNode> objectFieldNodes,
+            RuntimeConfig runtimeConfig,
+            Dictionary<string, HashSet<string>> fieldsToSupplyToReferencingEntities,
+            Dictionary<string, HashSet<string>> fieldsToDeriveFromReferencedEntities,
+            int nestingLevel,
+            IMetadataProviderFactory sqlMetadataProviderFactory)
+        {
+            foreach (ObjectFieldNode field in objectFieldNodes)
+            {
+                Tuple<IValueNode?, SyntaxKind> fieldDetails = GraphQLUtils.GetFieldDetails(field.Value, context.Variables);
+                SyntaxKind fieldKind = fieldDetails.Item2;
+
+                // For non-scalar fields, i.e. relationship fields, we have to recurse to process fields in the relationship field -
+                // which represents input data for a related entity.
+                if (fieldKind is not SyntaxKind.NullValue && !GraphQLUtils.IsScalarField(fieldKind))
+                {
+                    string relationshipName = field.Name.Value;
+                    string targetEntityName = runtimeConfig.Entities![entityName].Relationships![relationshipName].TargetEntity;
+                    HashSet<string>? derivedColumnsForEntity, columnsToBeDerivedFromEntity;
```

Comment:
```suggestion
                    HashSet<string>? columnValuePairsSourcedFromCurrentEntity, columnsToBeDerivedFromEntity;
```

- does `columnsToBeDerivedFromEntity` mean that the child entity must resolve values for these columns from within it's own input/autogenerated column values, or even a further nested "referenced" entity? Could this be named: `columnValuePairsToBeResolved`(+ `byChild`)?

Reply:
Your understanding is correct. Renaming variables as suggested.